agent/tcs/client/client.go

```diff
@@ -279,6 +281,48 @@ func (cs *clientServer) publishHealthMetricsOnce() error {
 	return nil
 }
 
+// publishInstanceHealthMetrics send the instance health information to backend
+func (cs *clientServer) publishInstanceHealthMetrics() {
+	if cs.publishTicker == nil {
+		seelog.Debug("Skipping publishing instance health metrics. Publish ticker is uninitialized")
+		return
+	}
+
+	// Publish instance metrics immediately after we connect and wait for ticks. This makes
+	// sure that there is no data loss when a scheduled metrics publishing fails
+	// due to a connection reset.
+	err := cs.publishInstanceHealthMetricsOnce()
+	if err != nil {
+		seelog.Warnf("Unable to publish instance health metrics: %v", err)
+	}
+	for {
+		select {
+		case <-cs.publishHealthTicker.C:
```

Comment:
Let's make a separate publishInstanceHealthTicker that takes it's own publishing interval. We will want this to be longer than the metrics publish interval because we need to make sure that we have sufficient data before sending. We will figure out an exact number through prototyping but let's start with 3 minutes.

Reply:
Done.